engines/efh/efh.cpp

```diff
@@ -0,0 +1,2668 @@
+					displayFctFullScreen();
+			}
+			Common::KeyCode input = waitForKey();
+			if (input == Common::KEYCODE_y) {
+				displayMenuAnswerString("-> Yes <-", 24, 296, 169);
+				getInput(2);
+				saveGameDialog();
+			} else {
+				displayMenuAnswerString("-> No!!! <-", 24, 296, 169);
+				getInput(2);
+			}
+			clearBottomTextZone_2(0);
+			displayLowStatusScreen(true);
+
+			}
+			break;
+		case Common::KEYCODE_F7: { // Original is using CTRL-L
+			for (uint counter = 0; counter < 2; ++counter) {
+				clearBottomTextZone(0);
+				displayCenteredString("Are You Sure You Want To Load?", 24, 296, 160);
+				if (counter == 0)
+					displayFctFullScreen();
+			}
+			Common::KeyCode input = waitForKey();
+			if (input == Common::KEYCODE_y) {
+				displayMenuAnswerString("-> Yes <-", 24, 296, 169);
+				getInput(2);
+				loadGameDialog();
+			} else {
+				displayMenuAnswerString("-> No!!! <-", 24, 296, 169);
+				getInput(2);
+			}
+			clearBottomTextZone_2(0);
+			displayLowStatusScreen(true);
+
+		} break;
+		// debug cases to test sound
+		case Common::KEYCODE_1:
+			generateSound(13);
+			break;
+		case Common::KEYCODE_2:
+			generateSound(14);
+			break;
+		case Common::KEYCODE_3:
+			generateSound(15);
+			break;
+		case Common::KEYCODE_4:
+			generateSound(5);
+			break;
+		case Common::KEYCODE_5:
+			generateSound(10);
+			break;
+		case Common::KEYCODE_6:
+			generateSound1(20, 888, 3000);
+			break;
+		default:
+			if (retVal != Common::KEYCODE_INVALID)
+				warning("Main Loop: Unhandled input %d", retVal);
+			break;
+		}
+
+		if ((_mapPosX != _oldMapPosX || _mapPosY != _oldMapPosY) && !_shouldQuit) {
+			bool collisionFl = checkMonsterCollision();
+			if (_dbgForceMonsterBlock || collisionFl) {
+				_oldMapPosX = _mapPosX;
+				_oldMapPosY = _mapPosY;
+				_oldImageSetSubFilesIdx = _imageSetSubFilesIdx;
+				_redrawNeededFl = true;
+			} else {
+				_mapPosX = _oldMapPosX;
+				_mapPosY = _oldMapPosY;
+				if (_oldImageSetSubFilesIdx != _imageSetSubFilesIdx) {
+					_redrawNeededFl = true;
+					_oldImageSetSubFilesIdx = _imageSetSubFilesIdx;
+				}
+			}
+			if (_largeMapFlag) {
+				_techDataId_MapPosX = _mapPosX;
+				_techDataId_MapPosY = _mapPosY;
+			}
+		}
+
+		if (!_shouldQuit) {
+			handleMapMonsterMoves();
+		}
+
+		if (_redrawNeededFl && !_shouldQuit) {
+			drawScreen();
+			displayLowStatusScreen(true);
+		}
+
+		if (!_shouldQuit) {
+			handleNewRoundEffects();
+
+			if (_tempTextDelay > 0) {
+				if (--_tempTextDelay == 0) {
+					displayMiddleLeftTempText(nullptr, true);
+				}
+			}
+		}
+
+		if (_unk2C8AA > 0)
+			--_unk2C8AA;
+
+		if (isTPK()) {
+			if (handleDeathMenu())
+				_shouldQuit = true;
+		}
+
+		displayFctFullScreen();
+	}
+	return Common::kNoError;
+}
+
+void EfhEngine::initialize() {
+	_rnd = new Common::RandomSource("Hell");
+	_rnd->setSeed(g_system->getMillis());   // Kick random number generator
+	_shouldQuit = false;
+}
+
+void EfhEngine::songDelay(int delay) {
+	debug("songDelay %ld", delay);
```

Comment:
Hide it behind `kDebugEngine` like in other places? Same for the methods below.

Reply:
Moved to sound.cpp + debug has been repaced